lib/logflare/backends/adaptor/tcp_adaptor/syslog.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Logflare.Backends.Adaptor.TCPAdaptor.Syslog do
+  @moduledoc """
+  Implementation of [RFC5424 The Syslog Protocol][]
+
+  It uses [octet-counting framing][].
+
+  [RFC5424]: https://www.rfc-editor.org/rfc/rfc5424
+  [octet-counting framing]: https://www.rfc-editor.org/rfc/rfc6587#section-3.4.1
+  """
+
+  alias Logflare.LogEvent
+
+  def format(log_event, cipher_key \\ nil) do
+    %LogEvent{id: id, body: body} = log_event
+
+    level = get_in(body["body"]["level"]) || body["level"] || "info"
+    pri = 16 * 8 + severity_code(level)
+
+    timestamp =
+      body
+      |> Map.fetch!("timestamp")
+      |> DateTime.from_unix!(:microsecond)
+      |> DateTime.truncate(:millisecond)
+      |> DateTime.to_iso8601()
+
+    metadata = body["metadata"]
+    hostname = get_in(metadata["host"]) || body["host"]
+    app_name = get_in(metadata["app_name"]) || body["app_name"] || "logflare"
+    procid = get_in(metadata["pid"]) || body["pid"]
```

Comment:
should factor in OpenTelemetry resource as well, which would be at the top-level `resource` field.
scope is also sometimes used but we can ignore that for now.

<img width="868" height="418" alt="CleanShot 2025-12-10 at 19 04 54@2x" src="https://github.com/user-attachments/assets/c212786d-74b1-4aa4-b642-074a611087df" />

Reply:
Added `resource` in https://github.com/Logflare/logflare/pull/2985/commits/22a0b0dd773ef22be19432bd7915e09b5646bdec